src/App.tsx

```diff
@@ -16,35 +16,101 @@
 ];
 
 export const App: React.FC = () => {
+  const [sort, setSort] = useState<string[]>([...goodsFromServer]);
+  const [reverse, setReverse] = useState<boolean>(false);
+  const [active, setActive] = useState<string>('');
+
+  const handleSort = (good: string): void => {
+    let newSort: string[];
+
+    switch (good) {
+      case 'abc':
+        if (reverse) {
+          newSort = [...sort].sort((a, b) => b.localeCompare(a));
+          setActive(good);
+        } else {
+          newSort = [...sort].sort((a, b) => a.localeCompare(b));
+          setActive(good);
+        }
+
+        break;
+      case 'length':
+        if (reverse) {
+          newSort = [...sort].sort((a, b) => b.length - a.length);
+          setActive(good);
+        } else {
+          newSort = [...sort].sort((a, b) => a.length - b.length);
+          setActive(good);
+        }
+
+        break;
+      case 'reverse':
+        if (active === 'abc' || active === 'length') {
+          setReverse(!reverse);
+        } else if (active === 'reverse') {
+          setReverse(!reverse);
+          setActive('');
+        } else {
+          setReverse(!reverse);
+          setActive('reverse');
+        }
```

Comment:
The logic for toggling the reverse state and setting the active state in the 'reverse' case is a bit convoluted. Consider simplifying it by directly toggling the reverse state and setting the active state based on the current state. This will make the code easier to understand and maintain.